wavesrv/pkg/cmdrunner/cmdrunner.go

```diff
@@ -1102,6 +1108,551 @@ func SidebarRemoveCommand(ctx context.Context, pk *scpacket.FeCommandPacketType)
+		sourceIds, err := resolveUiIds(ctx, pk, R_Remote)
+		if err != nil {
+			return nil, fmt.Errorf("error resolving remote id %v", err)
+		}
+		sourceRemoteId = sourceIds.Remote
+	}
+	destInfo := pk.Args[1]
+	destRemote, destPath := parseCopyFileParam(destInfo)
+	if destRemote == "error" {
+		return nil, fmt.Errorf("error: malformed arguments - usage: [remote]:path ")
+	} else if destRemote == "" {
+		destRemote = "connected"
+		destRemoteId = ids.Remote
+		if ids.Remote.DisplayName == "local" {
+			destRemote = "local"
+		}
+	} else {
+		pk.Kwargs["remote"] = destRemote
+		destIds, err := resolveUiIds(ctx, pk, R_Remote)
+		if err != nil {
+			return nil, fmt.Errorf("error resolving remote id %v", err)
+		}
+		destRemoteId = destIds.Remote
+	}
+	if destPath == "" {
+		return nil, fmt.Errorf("error: malformed arguments - usage: [remote]:path ")
+	}
+
+	var sourceFullPath string
+	var destFullPath string
+	sourceMsh := sourceRemoteId.MShell
+	if sourceMsh == nil {
+		return nil, fmt.Errorf("failure getting source remote mshell")
+	}
+	sourceRRState := sourceMsh.GetRemoteRuntimeState()
+	sourcePathWithHome, err := sourceRRState.ExpandHomeDir(sourcePath)
+	if err != nil {
+		return nil, fmt.Errorf("expand home dir err: %v", err)
+	}
+	sourceFullPath = sourcePathWithHome
+	if (sourceRemote == "connected" || sourceRemote == "local") && !filepath.IsAbs(sourcePathWithHome) && sourceRemoteId.FeState != nil {
+		sourceCwd := sourceRemoteId.FeState["cwd"]
+		if sourceCwd != "" {
+			sourceFullPath = filepath.Join(sourceCwd, sourcePathWithHome)
+		}
+	}
+	if destPath[len(destPath)-1:] == "/" {
+		sourceFileName := filepath.Base(sourceFullPath)
+		destPath = filepath.Join(destPath, sourceFileName)
+	}
+	destMsh := destRemoteId.MShell
+	if destMsh == nil {
+		return nil, fmt.Errorf("failure getting dest remote mshell")
+	}
+	destRRState := destMsh.GetRemoteRuntimeState()
+	destPathWithHome, err := destRRState.ExpandHomeDir(destPath)
+	if err != nil {
+		return nil, fmt.Errorf("expand home dir err: %v", err)
+	}
+	destFullPath = destPathWithHome
+	if (destRemote == "connnected" || destRemote == "local") && !filepath.IsAbs(destPathWithHome) && destRemoteId.FeState != nil {
+		destCwd := destRemoteId.FeState["cwd"]
+		if destCwd != "" {
+			destFullPath = filepath.Join(destCwd, destPathWithHome)
+		}
+	}
+	var outputPos int64
+	outputStr := fmt.Sprintf("Copying [%v]:%v to [%v]:%v\r\n", sourceRemoteId.DisplayName, sourceFullPath, destRemoteId.DisplayName, destFullPath)
+	termopts := sstore.TermOpts{Rows: shellutil.DefaultTermRows, Cols: shellutil.DefaultTermCols, FlexRows: true, MaxPtySize: remote.DefaultMaxPtySize}
+	cmd, err := makeDynCmd(ctx, "copy file", ids, pk.GetRawStr(), termopts)
+	writeStringToPty(ctx, cmd, outputStr, &outputPos)
+	if err != nil {
+		// TODO tricky error since the command was a success, but we can't show the output
+		return nil, err
+	}
+	update, err := addLineForCmd(ctx, "/copy file", false, ids, cmd, "", nil)
```

Comment:
been looking into renderers for the file view task - is having "renderer" set to "" the right thing here?

Reply:
ya, an empty string renderer defaults to "terminal"